File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Symbols/WrappedEventSymbol.cs
using System.Collections.Immutable;
using System.Globalization;
using System.Threading;

namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
{
	internal abstract class WrappedEventSymbol : EventSymbol
	{
		protected readonly EventSymbol _underlyingEvent;

		public EventSymbol UnderlyingEvent => _underlyingEvent;

		public override bool IsImplicitlyDeclared => _underlyingEvent.IsImplicitlyDeclared;

		internal override bool HasSpecialName => _underlyingEvent.HasSpecialName;

		public override string Name => _underlyingEvent.Name;

		public override ImmutableArray<Location> Locations => _underlyingEvent.Locations;

		public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => _underlyingEvent.DeclaringSyntaxReferences;

		public override Accessibility DeclaredAccessibility => _underlyingEvent.DeclaredAccessibility;

		public override bool IsShared => _underlyingEvent.IsShared;

		public override bool IsOverridable => _underlyingEvent.IsOverridable;

		public override bool IsOverrides => _underlyingEvent.IsOverrides;

		public override bool IsMustOverride => _underlyingEvent.IsMustOverride;

		public override bool IsNotOverridable => _underlyingEvent.IsNotOverridable;

		internal override ObsoleteAttributeData ObsoleteAttributeData => _underlyingEvent.ObsoleteAttributeData;

		public override bool IsWindowsRuntimeEvent => _underlyingEvent.IsWindowsRuntimeEvent;

		internal override bool HasRuntimeSpecialName => _underlyingEvent.HasRuntimeSpecialName;

		public WrappedEventSymbol(EventSymbol underlyingEvent)
		{
			_underlyingEvent = underlyingEvent;
		}

		public override string GetDocumentationCommentXml(CultureInfo preferredCulture = null, bool expandIncludes = false, CancellationToken cancellationToken = default(CancellationToken))
		{
			return _underlyingEvent.GetDocumentationCommentXml(preferredCulture, expandIncludes, cancellationToken);
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Symbols/WrappedEventSymbol.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports System
Imports System.Collections.Immutable
Imports System.Globalization
Imports System.Threading

Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
	Friend MustInherit Class WrappedEventSymbol
		Inherits EventSymbol
		Protected ReadOnly _underlyingEvent As EventSymbol

		Public Overrides ReadOnly Property DeclaredAccessibility As Accessibility
			Get
				Return Me._underlyingEvent.DeclaredAccessibility
			End Get
		End Property

		Public Overrides ReadOnly Property DeclaringSyntaxReferences As ImmutableArray(Of SyntaxReference)
			Get
				Return Me._underlyingEvent.DeclaringSyntaxReferences
			End Get
		End Property

		Friend Overrides ReadOnly Property HasRuntimeSpecialName As Boolean
			Get
				Return Me._underlyingEvent.HasRuntimeSpecialName
			End Get
		End Property

		Friend Overrides ReadOnly Property HasSpecialName As Boolean
			Get
				Return Me._underlyingEvent.HasSpecialName
			End Get
		End Property

		Public Overrides ReadOnly Property IsImplicitlyDeclared As Boolean
			Get
				Return Me._underlyingEvent.IsImplicitlyDeclared
			End Get
		End Property

		Public Overrides ReadOnly Property IsMustOverride As Boolean
			Get
				Return Me._underlyingEvent.IsMustOverride
			End Get
		End Property

		Public Overrides ReadOnly Property IsNotOverridable As Boolean
			Get
				Return Me._underlyingEvent.IsNotOverridable
			End Get
		End Property

		Public Overrides ReadOnly Property IsOverridable As Boolean
			Get
				Return Me._underlyingEvent.IsOverridable
			End Get
		End Property

		Public Overrides ReadOnly Property IsOverrides As Boolean
			Get
				Return Me._underlyingEvent.IsOverrides
			End Get
		End Property

		Public Overrides ReadOnly Property IsShared As Boolean
			Get
				Return Me._underlyingEvent.IsShared
			End Get
		End Property

		Public Overrides ReadOnly Property IsWindowsRuntimeEvent As Boolean
			Get
				Return Me._underlyingEvent.IsWindowsRuntimeEvent
			End Get
		End Property

		Public Overrides ReadOnly Property Locations As ImmutableArray(Of Location)
			Get
				Return Me._underlyingEvent.Locations
			End Get
		End Property

		Public Overrides ReadOnly Property Name As String
			Get
				Return Me._underlyingEvent.Name
			End Get
		End Property

		Friend Overrides ReadOnly Property ObsoleteAttributeData As Microsoft.CodeAnalysis.ObsoleteAttributeData
			Get
				Return Me._underlyingEvent.ObsoleteAttributeData
			End Get
		End Property

		Public ReadOnly Property UnderlyingEvent As EventSymbol
			Get
				Return Me._underlyingEvent
			End Get
		End Property

		Public Sub New(ByVal underlyingEvent As EventSymbol)
			MyBase.New()
			Me._underlyingEvent = underlyingEvent
		End Sub

		Public Overrides Function GetDocumentationCommentXml(Optional ByVal preferredCulture As CultureInfo = Nothing, Optional ByVal expandIncludes As Boolean = False, Optional ByVal cancellationToken As System.Threading.CancellationToken = Nothing) As String
			Return Me._underlyingEvent.GetDocumentationCommentXml(preferredCulture, expandIncludes, cancellationToken)
		End Function
	End Class
End Namespace